kotlin-asyncapi-context/src/main/kotlin/org/openfolder/kotlinasyncapi/context/annotation/AnnotationProvider.kt

```diff
@@ -24,14 +24,15 @@ import org.openfolder.kotlinasyncapi.model.server.ReferencableServerVariablesMap
 import org.openfolder.kotlinasyncapi.model.server.ReferencableServersMap
 import kotlin.reflect.KClass
 import kotlin.reflect.full.findAnnotation
+import kotlin.reflect.full.functions
 
 class AnnotationProvider(
     private val applicationPackage: Package? = null,
     private val classLoader: ClassLoader? = null,
     private val scanner: AnnotationScanner,
     private val messageProcessor: AnnotationProcessor<Message, KClass<*>>,
     private val schemaProcessor: AnnotationProcessor<Schema, KClass<*>>,
-    private val channelProcessor: AnnotationProcessor<Channel, KClass<*>>
+    private val channelProcessor: AnnotationProcessor<Channel, Any>
```

Comment:
I would like to keep it type safe. If we define a new component annotation processor we could keep the current setup. component processor would still be AnnotationProcessor<Schema, KClass<*>> and look for channel annotation on its functions ect.

Reply:
Yup I can split those out.

Reply:
This has been updated. Please LMK what you think.